integration/sql/impl/src/context/mod.rs

```diff
@@ -347,9 +347,28 @@ impl<'a> Context<'a> {
         for (col, ancestors) in &frame.column_ancestry {
             let mut expanded = ColumnAncestors::new();
             for ancestor in ancestors {
-                let prev = old.column_ancestry.get(ancestor);
-                if let Some(list) = prev {
-                    expanded.extend(list.iter().cloned());
+                // here we need also look on other if they are connected
+                let mut result = Vec::new();
+                let mut stack = Vec::new();
+
+                stack.push(ancestor.clone());
+
+                while let Some(current) = stack.pop() {
```

Comment:
This is quite dangerous - we've seen in the past infinite loops.

The example of the query (I know it's not ideal query but it's still correct) that hangs on the infinite loop:
```
WITH tab1 AS
  (SELECT s.col1,
          s.col2,
          s2.col3,
          s2.col4
   FROM stg AS s,
        stg2 AS s2),
     tab1 AS
  (SELECT col1,
          col2,
          col3,
          col4
   FROM tab1)
SELECT col1,
       col2,
       col3,
       col4
FROM tab1
```

Reply:
I may suggest `let mut visited = HashSet::new();` and skip on already visited?